static/js/src/app/settings.js

```diff
@@ -21,9 +22,229 @@ $(document).ready(function () {
+            document.documentElement.scrollTop = 0;
+            return false
+        }
+
+        var reshow = false
+        if ($("#advancedarea").is(":visible")) {
+            reshow = true
+        }
+        $("#advancedarea").hide("slow");
+        $("html, body").animate({ scrollTop: $(document).height() }, 1000);
+
+        var oldHTML = $("#validateimap").html();
+        // Disable inputs and change button text
+        $("#imaphost").attr("disabled", true);
+        $("#imapport").attr("disabled", true);
+        $("#imapusername").attr("disabled", true);
+        $("#imappassword").attr("disabled", true);
+        $("#use_imap").attr("disabled", true);
+        $("#use_tls").attr("disabled", true);
+        $("#validateimap").attr("disabled", true);
+        $("#validateimap").html("<i class='fa fa-circle-o-notch fa-spin'></i> Testing...");
+        
+        api.IMAP.validate(server).done(function(data) {
+            if (data.success == true) {
+                Swal.fire({
+                    title: "Success",
+                    html: "Logged into <b>" + $("#imaphost").val() + "</b>",
+                    type: "success",
+                })
+            } else {
+                Swal.fire({
+                    title: "Failed!",
+                    html: "Unable to login to <b>" + $("#imaphost").val() + "</b>.",
+                    type: "error",
+                    showCancelButton: true,
+                    cancelButtonText: "Close",
+                    confirmButtonText: "More Info",
+                    confirmButtonColor: "#428bca",
+                    allowOutsideClick: false,
+                }).then((result) => {
+                    Swal.fire({
+                        title: "Error:",
+                        text: data.message,
+                      })
+                  })
+            }
+            
+          })
+          .fail(function() {
+            Swal.fire({
+                title: "Failed!",
+                text: "An unecpected error occured.",
+                type: "error",
+            })
+          })
+          .always(function() {
+            //Re-enable inputs and change button text
+            $("#imaphost").attr("disabled", false);
+            $("#imapport").attr("disabled", false);
+            $("#imapusername").attr("disabled", false);
+            $("#imappassword").attr("disabled", false);
+            $("#use_imap").attr("disabled", false);
+            $("#use_tls").attr("disabled", false);
+            $("#validateimap").attr("disabled", false);
+            $("#validateimap").html(oldHTML);
+
+            if (reshow == true){
+                $("#advancedarea").show("slow");
+            }
+          });
+
+      }); //end testclick
+
+    $("#reporttab").click(function() {
+        loadIMAPSettings()
+    })
+
+    $("#advanced").click(function() {
+        $("#advancedarea").toggle("slow");
+        $("html, body").animate({ scrollTop: $(document).height() }, 1000);
```

Comment:
Could we avoid animating this? It's kind of jarring with how slow it is. My personal preference would be to change this to `toggle()` and remove the scroll animation following it. It's ok if someone clicks the link and the settings drop down below- the intuitive things, IMO, is that someone would then scroll to see the added settings that they can modify.

Reply:
Sure no problem! Done.